go/test/endtoend/cluster/topo_process.go

```diff
@@ -230,7 +230,7 @@ func (topo *TopoProcess) TearDown(Cell string, originalVtRoot string, currentRoo
 		// Attempt graceful shutdown with SIGTERM first
 		_ = topo.proc.Process.Signal(syscall.SIGTERM)
 
-		if !*keepData {
+		if !(*keepData || keepdata) {
```

Comment:
Why was this change needed? checking for nil?

Reply:
`keepData` is global and `keepdata` is passed a function parameter, we do not want the data to be deleted when we take down topo server inside the test.

Reply:
I now see the difference between the two. Terrible choice of variable names but we don't have to fix it in this PR.